ui-speedspacechart/src/components/layers/SpeedLimitTagsLayer.tsx

```diff
@@ -1,21 +1,39 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
 import type { Store, tooltipInfos } from '../../types/chartTypes';
 import Tooltip from '../common/Tooltip';
 import { LINEAR_LAYERS_HEIGHTS } from '../const';
-import { drawSpeedLimitTags, computeTooltip } from '../helpers/drawElements/speedLimitTags';
+import {
+  drawSpeedLimitTags,
+  computeTooltip,
+  loadSpeedLimitTagsImages,
+} from '../helpers/drawElements/speedLimitTags';
 
 type SpeedLimitTagsLayerProps = {
   width: number;
   marginTop: number;
   store: Store;
 };
+type ImagesType = {
+  questionImage: HTMLImageElement | null;
+  alertFillImage: HTMLImageElement | null;
+};
 const TOOLTIP_HEIGHT = 40;
 const MARGIN_ADJUSTMENT = 2;
 
 const SpeedLimitTagsLayer = ({ width, marginTop, store }: SpeedLimitTagsLayerProps) => {
   const canvas = useRef<HTMLCanvasElement>(null);
   const tooltip = useRef<tooltipInfos | null>();
+  const [images, setImages] = useState<ImagesType>({ questionImage: null, alertFillImage: null });
+
+  useEffect(() => {
+    if (images.questionImage !== null && images.alertFillImage !== null) return;
+    const fetchImages = async () => {
+      const { questionImage, alertFillImage } = await loadSpeedLimitTagsImages();
+      setImages({ questionImage, alertFillImage });
+    };
+    fetchImages();
+  }, [images]);
```

Comment:
Hm, this is a bit surprising: this `useEffect()` depends on `images` and also calls `setImages()`. Am I missing something here.

Reply:
The linter wanted me to add images to the dependency, I am not sure why, hence why I also explicitly added a check to skip if the images were already loaded.

The whole thing should disappear in the refactor though, it will then have imageurls as a dep, which will make more sense ^^

Reply:
Went further and put even the urls in the useeffect, with an empty dependence array, which should make the code flow as clear as possible